addons/mail/security/mail_security.xml

```diff
@@ -142,6 +142,33 @@
             <field name="perm_read" eval="False"/>
             <field name="perm_write" eval="False"/>
         </record>
+
+        <record id="ir_rule_ir_attachment_write_unlink" model="ir.rule">
+            <field name="name">ir.attachment: internal users can only delete own attachments</field>
+            <field name="model_id" ref="base.model_ir_attachment"/>
+            <field name="groups" eval="[Command.link(ref('base.group_user'))]"/>
+            <field name="domain_force">
+                [
+                    "|",
+                        ("res_model", "!=", "discuss.channel"),
+                        "|",
+                            ("message_ids.author_id", "in", user.partner_id.id),
+                            ("channel_id.self_member_id.channel_role", "in", ("admin", "owner")),
+                ]
+            </field>
+            <field name="perm_create" eval="False"/>
+            <field name="perm_read" eval="False"/>
+            <field name="perm_write" eval="True"/>
+            <field name="perm_unlink" eval="True"/>
+        </record>
+
+        <record id="ir_rule_ir_attachment_group_system" model="ir.rule">
+            <field name="name">ir_attachment: admin full access</field>
+            <field name="model_id" ref="base.model_ir_attachment"/>
+            <field name="groups" eval="[Command.link(ref('base.group_system'))]"/>
+            <field name="domain_force">[(1, "=", 1)]</field>
+        </record>
```

Comment:
To double check: is that actually working as expected?
As `_check_access` (and corresponding `_search`) is heavily overridden in the attachment model.

A lot of refactoring have been done recently, so I'm not sure exactly what is the current situation.

In any case, we should be adding tests in this PR to cover all possible cases.

Reply:
Would say to scrap everything and cleanly override ``_check_access``, if really necessary. Will take a lot less code.